main/src/ca/uwaterloo/flix/language/ast/Symbol.scala

```diff
@@ -546,6 +546,34 @@ object Symbol {
     def namespace: List[String] = enumSym.namespace :+ enumSym.name
   }
 
+  /**
+   * Struct Field Symbol.
+   */
+  final class StructFieldSym(val structSym: Symbol.StructSym, val name: String, val loc: SourceLocation) extends Symbol {
```

Comment:
This symbol may not be necessary. The Name.Field might be enough, but we can discuss later.